src/App.tsx

```diff
@@ -21,22 +31,37 @@ export const App: React.FC = () => {
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
+                {!selectedUser.user && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <Loader />
+                {isLoading && <Loader />}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {loadingPostError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {loadingPostError.errorText}
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {!isLoading &&
+                  !loadingPostError &&
+                  selectedUser.user &&
+                  userPosts.length === 0 && (
```

Comment:
```suggestion
                  !userPosts.length && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**